src/components/Reviews.jsx

```diff
@@ -4,27 +4,47 @@ import { Review } from "./Review";
 
 export function Reviews() {
   const [reviews, setReviews] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const [chosenCategory, setChosenCategory] = useState("all");
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
+    let url = "https://be-nc-games.onrender.com/api/reviews";
+    if (chosenCategory !== "all") {
+      url += `?category=${chosenCategory}`;
+    }
+
     axios
-      .get("https://be-nc-games.onrender.com/api/reviews")
+      .get(url)
       .then((response) => {
         setReviews(response.data.reviews);
         setIsLoading(false);
+
+        return axios.get("https://be-nc-games.onrender.com/api/categories");
+      })
+      .then((response) => {
+        const fetchedCategories = response.data.categories;
+        const fetchedCategoySlugs = fetchedCategories.map((category) => category.slug);
+        const newCategories = ["all", ...fetchedCategoySlugs];
+        setCategories(newCategories);
       })
       .catch((e) => {
         console.error(e);
         window.alert("ERROR: Could not contact the server, try again later");
       });
-  }, []);
+  }, [chosenCategory]);
 
   if (isLoading) {
     return <h2>Loading...</h2>;
   }
 
   return (
     <div className="Reviews">
+      <select name="Category" onChange={(e) => setChosenCategory(e.target.value)}>
+        {categories.map((category) => {
+          return <option value={category}>{category}</option>;
+        })}
+      </select>
```

Comment:
This should be extracted into a component dedicated to category drop down, it stops the bloating of this component.